### PR TITLE
wc-range-datepicker poc

### PR DESCRIPTION
**Pull request by jordanjones243**
# Alaska Airlines Pull Request

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

**Fixes:** #39

## Summary:

_Please summarize the scope of the changes you have submitted, what the intent of the work is and anything that describes the before/after state of the project._

## Type of change:

Please delete options that are not relevant.

- [x] New capability
- [ ] Revision of an existing capability
- [ ] Infrastructure change (automation, etc.)
- [ ] Other (please elaborate)

## Checklist:

- [ ] My update follows the CONTRIBUTING guidelines of this project
- [ ] I have performed a self-review of my own update

**By submitting this Pull Request, I confirm that my contribution is made under the terms of the Apache 2.0 license.**

_Pull Requests will be evaluated by their quality of update and whether it is consistent with the goals and values of this project. Any submission is to be considered a conversation between the submitter and the maintainers of this project and may require changes to your submission._

**Thank you for your submission!**<br>
-- Auro Design System Team

### CODE REVIEW

File: demo/demo.md [hunk 1]
```diff
@@ -16,14 +16,71 @@ The `<auro-calendar>` element should be used in situations where users may:
 
 ## Example(s)
 
-<div class="exampleWrapper">
-  <auro-calendar></auro-calendar>
-</div>
+### Basic Example
+<auro-calendar></auro-calendar>
```
Comment (jason-capsule42) [inline demo/demo.md:20]: `This` should be in the example wrapper.

```html
<div class="exampleWrapper">
  <auro-calendar></auro-calendar>
</div>
```

File: demo/demo.md [hunk 2]
```diff
@@ -16,14 +16,71 @@ The `<auro-calendar>` element should be used in situations where users may:
...
-</auro-accordion>
+</auro-accordion>
+
+### Force Narrow
+<auro-calendar forceNarrow></auro-calendar>
+<auro-accordion lowProfile justifyRight>
+  <span slot="trigger">See code</span>
+
+```html
+<auro-calendar forceNarrow></auro-calendar>
```
Comment (jason-capsule42) [inline demo/demo.md:36]: Let's discuss this part of the API more. I initial think the calendar should default to a single calendar and then we use an attribute to show additional month(s).

File: apiExamples/presetRange.html [hunk 3]
```diff
@@ -0,0 +1 @@
+<auro-calendar dateFrom="1677484800" dateTo="1678777200"></auro-calendar>
```
Comment (jason-capsule42) [inline apiExamples/presetRange.html:1]: I want to discuss what the right `type` is for these attibutes.

I think we should be expecting a Date object and then we parse that down to what is needed for the third party library.

File: demo/demo.md [hunk 4]
```diff
@@ -16,14 +16,71 @@ The `<auro-calendar>` element should be used in situations where users may:
...
+
+</auro-accordion>
+
+### Min and Max
+<auro-calendar month="08" year="2017" min="1504216800" max="1504908000"></auro-calendar>
+<auro-accordion lowProfile justifyRight>
+  <span slot="trigger">See code</span>
+
+```html
+<auro-calendar month="08" year="2017" min="1504216800" max="1504908000"></auro-calendar>
```
Comment (jason-capsule42) [inline demo/demo.md:69]: The examples should only use attributes necessary for the attribute(s) being explained. Is `month` and `year` necessary for `min` and `max` to work?

File: demo/demo.md [hunk 5]
```diff
@@ -16,14 +16,71 @@ The `<auro-calendar>` element should be used in situations where users may:
...
+
+</auro-accordion>
+
+### Custom Month
+<auro-calendar month="09"></auro-calendar>
+<auro-accordion lowProfile justifyRight>
+  <span slot="trigger">See code</span>
+
+```html
+<auro-calendar month="09"></auro-calendar>
```
Comment (jason-capsule42) [inline demo/demo.md:58]: Again I think we should use Date objects as much as possible. 

We can pass in a Date object to an attribute like `viewDate` and then auro-calendar can parse out the attributes that the 3rd party component needs.

File: demo/index.html [hunk 6]
```diff
@@ -41,14 +42,20 @@
...
     <script type="module" src="../apiExamples/selectedDate.js"></script>
-    <script type="module" src="../apiExamples/weekdayHeaderNotation.js"></script>
+    <script type="module" src="../apiExamples/weekdayHeaderNotation.js"></script> -->
 
     <!-- If additional elements are needed for the demo, add them here. -->
+    <script src="https://unpkg.com/@alaskaairux/auro-icon@latest/dist/auro-icon__bundled.js" type="module"></script>
     <script src="https://unpkg.com/@alaskaairux/auro-accordion@latest/dist/auro-accordion__bundled.js" type="module"></script>
+    <!-- <script> -->
+      <!-- console.log(new Date().getTime()); -->
+    <!-- </script> -->
```
Comment (jason-capsule42) [inline demo/index.html:59]: Commented out code needs to be removed.

File: demo/index.html [hunk 7]
```diff
@@ -41,14 +42,20 @@
...
+    <script type="module" src="../node_modules/wc-range-datepicker/dist/src/range-datepicker-calendar.js"></script>
+    <script type="module" src="../src/auro-calendar-month.js"></script>
 
-    <script type="module" src="../apiExamples/centralDate.js"></script>
+    <!-- <script type="module" src="../apiExamples/centralDate.js"></script>
     <script type="module" src="../apiExamples/maxDate.js"></script>
     <script type="module" src="../apiExamples/minDate.js"></script>
     <script type="module" src="../apiExamples/selectedDate.js"></script>
-    <script type="module" src="../apiExamples/weekdayHeaderNotation.js"></script>
+    <script type="module" src="../apiExamples/weekdayHeaderNotation.js"></script> -->
```
Comment (jason-capsule42) [inline demo/index.html:52]: Commented out code needs to be removed.

File: demo/index.html [hunk 8]
```diff
@@ -41,14 +42,20 @@
         });
     </script>
     <script type="module" src="../src/auro-calendar.js"></script>
+    <script type="module" src="../node_modules/wc-range-datepicker/dist/src/range-datepicker-calendar.js"></script>
```
Comment (jason-capsule42) [inline demo/index.html:45]: Is this needed? Should the script below already be importing this?

We definitely wouldn't want a user to import a 3rd party tool directly like this.

File: src/auro-calendar-month.js [hunk 9]
```diff
@@ -0,0 +1,82 @@
...
+  updated(changedProperties) {
+    if (changedProperties.has('year') || changedProperties.has('month')) {
+      this.yearAndMonthChanged(this.year, this.month);
+    }
+  }
+}
+// define the name of the custom component
+if (!customElements.get("auro-calendar-month")) {
+  customElements.define("auro-calendar-month", AuroCalendarMonth);
+}
```
Comment (jason-capsule42) [inline src/auro-calendar-month.js:82]: This should be happening in the root folder ./index.js like we do in hyperlink.

File: src/auro-calendar-month.js [hunk 10]
```diff
@@ -0,0 +1,82 @@
...
+ * - 'narrow' (e.g., T).
+ * Default is 'short'.
+ * @fires auroCalendar-dateSelected - Notifies that a date has been selected in the calendar.
+ * @fires auroCalendar-ready - Notifies that the component has finished initializing.
+ */
+
+/* eslint-disable no-self-assign */
+
+// class AuroCalendar extends LitElement {
+class AuroCalendarMonth extends RangeDatepickerCalendar {
```
Comment (jason-capsule42) [inline src/auro-calendar-month.js:29]: This needs to be converted to an export like hyperlink.

File: src/auro-calendar.js [hunk 11]
```diff
@@ -23,89 +26,105 @@ import styleCss from "./style-css.js";
 /* eslint-disable no-self-assign */
 
 // class AuroCalendar extends LitElement {
-class AuroCalendar extends LionCalendar {
+class AuroCalendar extends RangeDatepicker {
```
Comment (jason-capsule42) [inline src/auro-calendar.js:29]: This needs to be an export like hyperlink.

File: src/style.scss [hunk 12]
```diff
@@ -175,3 +175,18 @@
...
+}
+
+#container {
+  display: flex;
+  flex-direction: row;
+}
+
+#firstDatePicker {
+  margin-right: 16px;
+}
```
Comment (jason-capsule42) [inline src/style.scss:192]: I would make this selector be a left margin applied to `:not(first....)`

**Review by jason-capsule42 — CHANGES_REQUESTED** [review]

**Comment by jordanjones243** [comment]
Reference this ticket for further work for a Range Datepicker POV:
- https://github.com/AlaskaAirlines/auro-datepicker/issues/106

**Pull request closed without merging**